### PR TITLE
Issue #9236: Resolve 'new' suggesting wrong indentation for try handler

### LINKED ISSUE #9236: Regression in IndentationCheck 8.40 regarding the white spaces needed in a second line of a try/resource declaration
https://checkstyle.org/config_misc.html#Indentation

$ javac Test.java

successful compilation

$ Test.java

```java
import java.io.*;

class Test {
    void test() {
        try (InputStreamReader reader = 
            new InputStreamReader(System.in)) {         // violation
                // try block...    
        } catch (Exception e) {
            // catch block...
        }
    }
}
```

$ config.java

```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="Indentation" />
    </module>
  </module>
</module>
```

Actual

$ java -jar checkstyle-8.40-all.jar -c config.xml t2.java

```
[ERROR] /home/abhishhek/playground/Test.java:6:13: 
  'operator new' has incorrect indentation level 12, expected level should be 16. [Indentation]
```

Expected

No violations

-----

#9071 seems to have caused an unexpected side effect in the default behavior of IndentationCheck.

The following line used to be valid with Checkstyle 8.37: https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-13.0/xwiki-platform-core/xwiki-platform-icon/xwiki-platform-icon-default/src/main/java/org/xwiki/icon/internal/DefaultIconSetManager.java#L129

and is now failing with

> 'new' has incorrect indentation level 16, expected level should be 20.

My configuration is https://github.com/xwiki/xwiki-commons/blob/xwiki-commons-13.0/xwiki-commons-tools/xwiki-commons-tool-verification-resources/src/main/resources/checkstyle.xml#L205

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
Resolves #9236 
Diff Regression projects: https://gist.githubusercontent.com/Abhishek-kumar09/0745ecf11934e0214b9187f9d4896e4a/raw/a0cbb3d537560f88f699e778b41ffd3c8e1c9338/gistfile1.txt
Diff Regression config: https://gist.githubusercontent.com/Abhishek-kumar09/0745ecf11934e0214b9187f9d4896e4a/raw/a0cbb3d537560f88f699e778b41ffd3c8e1c9338/my_check.xml

### CODE REVIEW

**Comment by Abhishek-kumar09** [comment]
Github, generate report

**Review by strkkk — APPROVED** [review]

**Comment by romani** [comment]
this Pr depends on https://github.com/checkstyle/checkstyle/pull/9243 , after merge of such PR we should rebase and activate usage of SNAPSHOT version in xwiki validation to confirm fix and no other regressions.

**Review by romani — APPROVED** [review]
reports are good.
testing on guava code it better to be done with base=2 (for future)

**Review by romani — CHANGES_REQUESTED** [review]
please rebase on latest master
fix https://github.com/checkstyle/checkstyle/blob/db7215dc8b80fa99c27250c0548e9581f0f40208/.ci/travis/travis.sh#L271 to us snaptsht

**Comment by strkkk** [comment]
Github, rebase

**Comment by strkkk** [comment]
@Abhishek-kumar09 github action rebase will not work because of changes in github actions in master, please rebase manually

**Comment by Abhishek-kumar09** [comment]
Thanks for approving, though it doesn't look good to me: https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/issue_temp_2021155230/reports/diff/Orekit/index.html#A1

when I viewed it again, if you feel the same.

I think it should raise an error since, the line should be 4 levels ahead.

**Comment by romani** [comment]
@Abhishek-kumar09 , please rebase first to make sure that xwiki issue is resolved.

what you points too is good catch, BUT we can resolve it in separate issue. Because false-positve (as it is in xwiki) is big pain in comparison to false-negative (loss of violation). We can iteratively resolve problems. I am fine sometime when we do regression in behavior that there are some false-negatives. But I almost never allow regression with false-positives.

File: .ci/travis/travis.sh [hunk 1]
```diff
@@ -268,7 +268,7 @@ no-error-xwiki)
   # non-snapshot version until https://github.com/checkstyle/checkstyle/issues/9236
   # CS_POM_VERSION=$(mvn -e -q -Dexec.executable='echo' -Dexec.args='${project.version}' \
   #                   --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
-  CS_POM_VERSION=8.39
+  CS_POM_VERSION=8.41-SNAPSHOT
```
Comment (romani) [inline .ci/travis/travis.sh:271]: Please remove this line and uncomment previous

**Review by romani — CHANGES_REQUESTED** [review]
Item:

File: .ci/travis/travis.sh (hunk 1 above)
Comment (Abhishek-kumar09) [reply .ci/travis/travis.sh]: Sorry, Its done.

**Comment by Abhishek-kumar09** [comment]
Github, generate report

File: .ci/travis/travis.sh [hunk 2]
```diff
@@ -265,10 +265,8 @@ no-error-test-sbe)
   ;;
 
 no-error-xwiki)
-  # non-snapshot version until https://github.com/checkstyle/checkstyle/issues/9236
-  # CS_POM_VERSION=$(mvn -e -q -Dexec.executable='echo' -Dexec.args='${project.version}' \
-  #                   --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
-  CS_POM_VERSION=8.39
+   CS_POM_VERSION=$(mvn -e -q -Dexec.executable='echo' -Dexec.args='${project.version}' \
```
Comment (romani) [inline .ci/travis/travis.sh:268]: restore indentation (remove single space).

we do not have Check for this, but indentation matter.

**Review by romani — CHANGES_REQUESTED** [review]
> Guava on google style

https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#basic-difference-report-with-label-and-custom-projects-list

item:

File: .ci/travis/travis.sh (hunk 2 above)
Comment (Abhishek-kumar09) [reply .ci/travis/travis.sh]: Sorry again, mind jumbled up!
Corrected.

**Review by romani — APPROVED** [review]
Ok to merge

**Pull request merged**